### PR TITLE
feat(a2-1749): generate representation table

### PR DESCRIPTION
**Pull request by topickering**
## Ticket Number

https://pins-ds.atlassian.net/browse/A2-1749

## Description of change

Add Representation table to db schema

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/database/src/schema.prisma [hunk 1]
```diff
@@ -1205,3 +1205,25 @@ model Rule6Party {
...
+  /// this ID is for front office only
+  id String @id @default(dbgenerated("newid()")) @db.UniqueIdentifier
+
+  representationId        String /// The unique identifier for the representation
+  caseId                  Int? /// Internal case identifier
+  caseReference           String    @unique /// External case identifier
+  status                  String? /// Status of the representation, [ "awaiting_review", "referred", "valid", "invalid", "published", "archived", "draft", "withdrawn", null ]
+  originalRepresentation  String? /// The original representation
+  redacted                Boolean? /// Indicates if the representation is redacted
+  redactedRepresentation  String? /// The redacted version of the representation
```
Comment (beejjacobs) [inline packages/database/src/schema.prisma:1221]: FYI `String` is `nvarchar(1000)` (https://www.prisma.io/docs/orm/reference/prisma-schema-reference#string) - do we need to allow for longer representations here?

Comment (topickering) [reply packages/database/src/schema.prisma:1221]: That's a good point, I think currently we have that limitation in place on the submission side (this table is for the reps once they have been broadcast by BO), so we might need to standardise this across the codebase (including BO)

Comment (topickering) [reply packages/database/src/schema.prisma:1223]: Reverted as per our discussion and wider exercise looking at character limits

File: packages/database/src/schema.prisma [hunk 2]
```diff
@@ -1205,3 +1206,26 @@ model Rule6Party {
...
   AppealCase AppealCase? @relation(fields: [caseReference], references: [caseReference])
 }
+
+/// Proofs of Evidence, Final Comments, Statements, Planning Obligations and Interested Party Comments
+/// Received from BO will share this model and table
+model Representation {
+  /// this ID is for front office only
+  id String @id @default(dbgenerated("newid()")) @db.UniqueIdentifier
+
+  representationId        String /// The unique identifier for the representation
```
Comment (beejjacobs) [inline packages/database/src/schema.prisma:1216]: Should this be `@unique`?

File: packages/database/src/schema.prisma [hunk 3]
```diff
@@ -1205,3 +1206,26 @@ model Rule6Party {
...
+  caseId                  Int? /// Internal case identifier
+  caseReference           String     @unique /// External case identifier
```
Comment (beejjacobs) [inline packages/database/src/schema.prisma:1218]: Can this be unique? Can there not be multiple representations for one case?

Comment (topickering) [reply packages/database/src/schema.prisma:1218]: Good spots, got my uniques in the wrong place!

**Pull request merged**